sdk/ai/azure-ai-inference/azure/ai/inference/models/_patch.py

```diff
@@ -7,18 +7,58 @@
 Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
 """
 import asyncio
+import base64
 import json
 import logging
 import queue
 import re
+import sys
 
-from typing import List, AsyncIterator, Iterator
+from typing import List, AsyncIterator, Iterator, Optional, Union
 from azure.core.rest import HttpResponse, AsyncHttpResponse
+from ._models import ImageUrl as ImageUrlGenerated
 from .. import models as _models
 
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
 logger = logging.getLogger(__name__)
 
 
+class ImageUrl(ImageUrlGenerated):
+
+    @classmethod
+    def load(
+        cls,
+        *,
+        image_file: str,
+        image_format: str,
+        detail: Optional[Union[str, "_models.ImageDetailLevel"]] = None
+    ) -> Self:
+        """
+        Create an ImageUrl object from a local image file. The method reads the image
+        file and encodes it as a base64 string, which together with the image format
+        is then used to format the JSON `url` value passed in the request payload.
+
+        :ivar image_file: The name of the local image file to load. Required.
+        :vartype image_file: str
+        :ivar image_format: The MIME type format of the image. For example: "jpeg", "png". Required.
+        :vartype image_format: str
+        :ivar detail: The evaluation quality setting to use, which controls relative prioritization of
+         speed, token consumption, and accuracy. Known values are: "auto", "low", and "high".
+        :vartype detail: str or ~azure.ai.inference.models.ImageDetailLevel
+        :return: An ImageUrl object with the image data encoded as a base64 string.
+        :rtype: ~azure.ai.inference.models.ImageUrl
+        :raises FileNotFoundError when the image file could not be opened.
+        """
+        with open(image_file, "rb") as f:
```

Comment:
The main issue with opening a file inside a method is that it is a sync IO operation. And there is no async file `open` in the standard python library/builtins. We can either move the file open outside the method (and ultimately end up supporting both a sync and async file) or we can add a `async def aload(...)` method in the future.

I think the latter is fine (e.g. the code as written doesn't dig ourselves into too deep of a hole). But @annatisch , @kashifkhan , not sure if you have opinions on this?

Reply:
Per phone call, we can continue discussing this and make fixes as needed in another PR.